### PR TITLE
use `RandomState::with_seeds` replace `AHasher::default`

### PR DESCRIPTION
**Pull request by axxop**
Fix: #2093

AHash 0.8 related: https://github.com/tkaitchuck/aHash/issues/132

### CODE REVIEW

File: crates/egui/src/id.rs [hunk 1]
```diff
@@ -45,16 +45,16 @@ impl Id {
 
     /// Generate a new [`Id`] by hashing some source (e.g. a string or integer).
     pub fn new(source: impl std::hash::Hash) -> Id {
-        use std::hash::Hasher;
-        let mut hasher = epaint::ahash::AHasher::default();
+        use std::hash::{BuildHasher, Hasher};
+        let mut hasher = epaint::ahash::RandomState::with_seeds(15, 26, 37, 48).build_hasher();
```
Comment (Mingun) [inline crates/egui/src/id.rs:49]: Why exactly that seed is used? It is important? It definitely needs a comment. Also it is repeated three times, it is easy to change some of the places accidentally, which can break something, that is non-obvious from the code. I think, a function that will return `RandomState` is needed

Comment (emilk) [reply crates/egui/src/id.rs:49]: Let's use a more clear [nothing-up-my-sleeve number](https://en.wikipedia.org/wiki/Nothing-up-my-sleeve_number) like `1, 2, 3, 4`

Comment (emilk) [inline crates/egui/src/id.rs:49]:
```suggestion
        use std::hash::{BuildHasher as _, Hasher as _};
        let mut hasher = epaint::ahash::RandomState::with_seeds(1, 2, 3, 4).build_hasher();
```

File: crates/egui/src/id.rs [hunk 2]
```diff
@@ -45,16 +45,16 @@ impl Id {
...
         source.hash(&mut hasher);
         Id(hasher.finish())
     }
 
     /// Generate a new [`Id`] by hashing the parent [`Id`] and the given argument.
     pub fn with(self, child: impl std::hash::Hash) -> Id {
-        use std::hash::Hasher;
-        let mut hasher = epaint::ahash::AHasher::default();
+        use std::hash::{BuildHasher, Hasher};
+        let mut hasher = epaint::ahash::RandomState::with_seeds(15, 26, 37, 48).build_hasher();
```
Comment (emilk) [inline crates/egui/src/id.rs:57]:
```suggestion
        use std::hash::{BuildHasher as _, Hasher as _};
        let mut hasher = epaint::ahash::RandomState::with_seeds(1, 2, 3, 4).build_hasher();
```

File: crates/epaint/src/util/mod.rs [hunk 3]
```diff
@@ -5,10 +5,8 @@ pub use ordered_float::*;
 /// Hash the given value with a predictable hasher.
 #[inline]
 pub fn hash(value: impl std::hash::Hash) -> u64 {
-    use std::hash::Hasher as _;
-    let mut hasher = ahash::AHasher::default();
-    value.hash(&mut hasher);
-    hasher.finish()
+    use ahash::RandomState;
+    RandomState::with_seeds(15, 26, 37, 48).hash_one(value)
```
Comment (emilk) [inline crates/epaint/src/util/mod.rs:9]:
```suggestion
    use ahash::RandomState as _;
    RandomState::with_seeds(1, 2, 3, 4).hash_one(value)
```

**Review by emilk — CHANGES_REQUESTED** [review]

**Comment by emilk** [comment]
Good catch!

**Comment by axxop** [comment]
updated, please check

**Review by emilk — APPROVED** [review]
Thanks!

File: crates/egui/src/id.rs (hunk 1 above)
Comment (Mingun) [reply crates/egui/src/id.rs]: I think, it is also need to mention in a comment, that the selected numbers are just random. Should they need to be the same in all three places? Or it is just a [random values](https://xkcd.com/221/), not connected to each other? What's happened if in the future someone change the one of them to another values?

**Pull request merged**